source/views/components/filter/filter-popover.js

```diff
@@ -0,0 +1,65 @@
+//@flow
+import * as React from 'react'
+import {StyleSheet, View} from 'react-native'
+import Popover from 'react-native-popover-view'
+import {FilterSection} from './section'
+import type {FilterType} from './types'
+import * as c from '../colors'
+
+type Props = {
+	anchor: any,
+	filter: FilterType,
+	onClosePopover: (filter: FilterType) => any,
+	visible: boolean,
+}
+
+type State = {
+	filter: FilterType,
+}
+
+export class FilterPopover extends React.PureComponent<Props, State> {
+	state = {
+		filter: this.props.filter,
+	}
+
+	static getDerivedStateFromProps(props: Props) {
+		return {
+			filter: props.filter,
+		}
+	}
+
+	onFilterChanged = (filter: FilterType) => {
+		this.setState(() => ({filter: filter}))
+	}
+
+	render() {
+		const {filter} = this.state
+		const {anchor, onClosePopover, visible} = this.props
+		return (
+			<Popover
+				arrowStyle={arrowStyle}
+				fromView={anchor}
+				isVisible={visible}
+				onClose={() => {
+					onClosePopover(filter)
+				}}
+				placement="bottom"
+			>
+				<View style={styles.popoverContainer}>
+					<FilterSection filter={filter} onChange={this.onFilterChanged} />
+				</View>
+			</Popover>
+		)
+	}
+}
+
+const styles = StyleSheet.create({
+	arrow: {
+		backgroundColor: c.iosLightBackground,
+	},
+	popoverContainer: {
+		maxWidth: 300,
+	},
+})
+
+const arrowStyle = StyleSheet.flatten(styles.arrow)
```

Comment:
Why do you need to flatten the style?

Reply:
Weird bug in the `Popover` component...This is what we get without flattening:
<img width="308" alt="screen shot 2018-08-08 at 9 57 39 pm" src="https://user-images.githubusercontent.com/24640726/43875477-2e20ffd0-9b56-11e8-99fa-fbb1ff914835.png">

Reply:
Will do!